models/userStatus.js

```diff
@@ -17,7 +17,65 @@ const { TASK_STATUS } = require("../constants/tasks");
 const userStatusModel = firestore.collection("usersStatus");
 const tasksModel = firestore.collection("tasks");
 const { userState } = require("../constants/userStatus");
+const { getGroupRole, addGroupRoleToMember, removeGroupRoleFromMember } = require("./discordactions");
 const usersCollection = firestore.collection("users");
+const DISCORD_BASE_URL = config.get("services.discordBot.baseUrl");
+const jwt = require("jsonwebtoken");
+
+const removeGroupRoleFromDiscordUser = async ({ userId, roleName }) => {
```

Comment:
roleid property of the dataForDiscord object is not being set correctly.

The removeGroupRoleFromDiscordUser function takes a userId and roleName parameter and removes the role with the given name from the user's Discord account. The function first calls the getGroupRole function to get the role data, and then calls the removeGroupRoleFromMember function to remove the role from the user's Discord guild. Finally, the function calls the fetch function to make a DELETE request to the Discord API to remove the role from the user's Discord account.

Reply:
First, it removes the role data from our firestore database.
Yes, getGroupRole is for getting the ID of the role.
removeGroupRoleFromMember is for removing the role from our firestore database.
fetch call is for deleting it from the user on Discord.